main.py

```diff
@@ -69,23 +86,91 @@ def draw() -> Shape:
     return target_shape
 
 
+def draw_second_stage() -> Shape:
+    """Draw the next round screen and return the chosen shape"""
+    pygame.draw.rect(SCREEN, (0, 0, 0), (0, 0, 800, 600))
+
+    random.shuffle(shapes)
+    for index, shape in enumerate(shapes):
+        if index > 0:
+            break
+
+        position = shape.map(positions[index])
+
+        # TODO: Fix the issue with pygame and being able to draw the shape from the shape class
+        # TODO: This is a dirty hack because of the 24 hour deadline from start to finish
+        if shape.method == 'rect':
+            pygame.draw.rect(SCREEN, shape.colour, position)
+        elif shape.method == 'circle':
+            pygame.draw.circle(SCREEN, shape.colour, position, shape.radius)
+        elif shape.method == 'polygon':
+            pygame.draw.polygon(SCREEN, shape.colour, position)
+        elif shape.method == 'line':
+            pygame.draw.line(SCREEN, shape.colour, position[0], position[1], shape.radius)
+
+    target_shape = shapes[random.randint(0, 3)]
+    while target_shape.method == last_shape:
+        target_shape = shapes[random.randint(0, 3)]
+
+    font = pygame.font.Font(None, 48)
+    text = font.render(f"{current_round}. Please select the {target_shape.type}", True, (255, 255, 255))
+    text_rect = text.get_rect(center=(400, 50))
+    SCREEN.blit(text, text_rect)
+
+    return target_shape
+
+
+def draw_third_stage() -> Shape:
+    """Draw the next round screen and return the chosen shape"""
+    pygame.draw.rect(SCREEN, (0, 0, 0), (0, 0, 800, 600))
+    path = 'D:\pedro\Documents\Tese\data_collect_py\IAPS/*.jpg'
+    IAPS = []
+    for filename in glob.glob(path):
+        IAPS.append(filename)
+    figure = random.choice(IAPS)
+    if figure in picked_IAPS:
+        while figure in picked_IAPS:
+            figure = random.choice(IAPS)
+    image = pygame.image.load(figure)
+    SCREEN.fill(0, 0, 0)
+    SCREEN.blit(image, (0, 0))
+    return figure
+
+
+def draw_whitespace():
+    """Draw the next round screen and return the chosen shape"""
+    pygame.draw.rect(SCREEN, (0, 0, 0), (0, 0, 800, 600))
+    SCREEN.fill(255, 255, 255)
+
+
+print('5')
+
 shapes = load_shapes()
 positions = ["top_left", "top_right", "bottom_left", "bottom_right"]
 
+print('6')
+
 pygame.init()
 SCREEN = pygame.display.set_mode((800, 600))
 
+print('7')
+
 # TODO: Build structure / classes to handle events better
 generate_new_round = pygame.USEREVENT + 1
 pygame.time.set_timer(generate_new_round, 5000)
 
-max_rounds = 24
-current_round = 1
+print('8')
 
+max_rounds = 81
```

Comment:
I'm not keen on this, is there any mathematical equation or pattern we can follow to clean this up?

Reply:
Are we even using that variable, I only filled it in with my own calculation because you put it there, but I do not understand where we are using it

Reply:
But yeah, I can work that out

Reply:
@pedrorribeiro Yeah looks like it is deprecated, I can't recall if I used it in my 0.1.0 release.

Reply:
It has become a good idea again if we can dynamically calculate the number of rounds and plop it back into the while loop, right?